### PR TITLE
Fix/payg overage usage type

### PR DESCRIPTION
**Pull request by aphethean1**
This PR addresses [issue#23](https://github.com/get-select/dbt-snowflake-monitoring/issues/23)

### CODE REVIEW

File: models/cost_per_query.sql [hunk 1]
```diff
@@ -6,7 +6,7 @@ daily_rates as (
         date,
         effective_rate
     from {{ ref('daily_rates') }}
-    where usage_type = 'compute'
+    where (usage_type = 'compute' or usage_type = 'overage-compute')
```
Comment (NiallRees) [inline models/cost_per_query.sql:9]:
```suggestion
    where usage_type in ('compute', 'overage-compute')
    group by date
```

This concerns me a little in the event that there's both a `compute` and an `overage-compute` entry for a day. I have no idea if that can happen. For now, let's just group by `date` and take the max rate?

File: models/cost_per_query.sql [hunk 2]
```diff
@@ -6,7 +6,7 @@ daily_rates as (
         date,
         effective_rate
```
Comment (NiallRees) [inline models/cost_per_query.sql:7]:
```suggestion
        max(effective_rate) as effective_rate
```

File: models/daily_spend.sql [hunk 3]
```diff
@@ -81,7 +81,7 @@ storage_terabytes_daily as (
     left join storage_terabytes_daily on dates.date = storage_terabytes_daily.date
     left join {{ ref('daily_rates') }} on
         storage_terabytes_daily.date = daily_rates.date
-        and daily_rates.usage_type = 'storage'
+        and (daily_rates.usage_type = 'storage' or daily_rates.usage_type = 'overage-storage')
```
Comment (NiallRees) [inline models/daily_spend.sql:84]: We can either do a group by here (by date, service, storage type, warehouse_name, database_name), or do it in a CTE just on date and join it in.

**Comment by NiallRees** [comment]
Thanks @aphethean1! Just a comment about a potential fanout. For now, let's just ensure that there's only ever a single row by grouping on the date if that sounds good to you.

File: models/cost_per_query.sql (hunk 1 above)
Comment (aphethean1) [reply models/cost_per_query.sql:9]: Eeish.  Good point.  My understanding is that 100% of spend is overage on payg plans, on a committed plan the overage will start accruing after the agreed spend has been used up.  I guess that means they will both happen on the same day for some clients.

File: models/daily_spend.sql (hunk 3 above)
Comment (aphethean1) [reply models/daily_spend.sql:84]: I've added the group by (like the other spend types) and the models have run successfully.  Full disclosure, we have no storage spend of these types and no tests so feels like a bit of a hit and hope commit.  https://github.com/aphethean1/dbt-snowflake-monitoring/commit/17379120f6ecafce9eaf7a686dbe88df1b69c06e

**Pull request merged**